contracts/modules/cover/Cover.sol

```diff
@@ -425,7 +425,12 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
         require(ok, "Cover: Sending ETH to commission destination failed.");
       }
 
-      // TODO: send eth to pool
+      // send premium in eth to the pool
+      if (premiumInPaymentAsset > 0) {
```

Comment:
The `premiumInPaymentAsset` is already checked at line 294 so the if statement here seems redundantant. Please check and confirm.

Secondly, I'm wondering whether it's worth moving this transfer before sending the remainder to the `msg.sender`. Thinking of re-entrnacy here but I think it's very unlikely to do anything realistic here. Thoughts? cc @danoctavian

Reply:
Yup, the if statement seems to be redundant, will remove it.. The second point seems like a good idea to me, even if the risk is low.

Reply:
@shark0der we used the nonReentrant guard for buyCover to allow flexibility in how we structure code in `buyCover` and prioritize avoiding stack to deep and readability instead of preventing re-entrancy attacks through ordering.

I think nonReentrant guard is sufficient